src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -296,7 +296,71 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
     for (auto& img : images)
         img->drop();
     return SAssetBundle({core::smart_refctd_ptr<IAsset>(tex, core::dont_grab)});
+
+
+
 #else
+	core::vector<asset::ICPUImage*> images;
+
+	switch (header.ImageType)
+	{
+	case 1: // uncompressed color-mapped image
+	case 2: // uncompressed rgb image
+	case 3: // uncompressed grayscale image
+	case 10:
+	{
+		const uint64_t imageSize = header.ImageHeight * header.ImageWidth * header.PixelDepth / 8;
+		void* data = new uint8_t[imageSize];
+		core::smart_refctd_ptr<ICPUBuffer> imageBuffer(new ICPUBuffer(imageSize));
```

Comment:
this will leak (double grab), use `auto imageBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(imageSize);` instead